algorithms/src/snark/varuna/data_structures/proof.rs

```diff
@@ -378,6 +380,88 @@ impl<E: PairingEngine> FromBytes for Proof<E> {
     }
 }
 
+/// Predicts the size in bytes of a Varuna proof as produced by
+/// `Proof::serialize_compressed` without needing to receive the proof itself.
+///
+/// *Arguments*:
+///  - `batch_sizes`: the batch sizes of the circuits and instances being
+///    proved.
+///  - `hiding`: indicates whether the proof system is run in ZK mode
+///  - `varuna_version`: the version of Varuna being used
+///
+/// *Returns*:
+///  - `Some(size)` for `VarunaVersion::V2`, where `size` is the size of the
+///    proof in bytes.
+///  - `None` for `VarunaVersion::V1`.
+pub fn proof_size<E: PairingEngine>(
+    batch_sizes: &[usize],
+    varuna_version: VarunaVersion,
+    hiding: bool,
+) -> Option<usize> {
+    let n_circuits: usize = batch_sizes.len();
+    let n_instances: usize = batch_sizes.iter().sum();
+
+    match varuna_version {
+        VarunaVersion::V1 => None,
+        VarunaVersion::V2 => {
+            // All fields are serialised in Compressed mode The breakdown is as
```

Comment:
```suggestion
            // All fields are serialized in `Compressed` mode. The breakdown is as
```
As for the nits on spelling for "serialized" and quotation for code components, they're very minor. Feel free to take or leave.